_posts/2022-02-07-feature-disentanglement1.md

```diff
@@ -0,0 +1,131 @@
+
+
+# Disentanglement
+Feature disentanglement is isolating the source of variation in observation data. There is a lot more factors/feature of an MNIST image other than the number itself, such as the location of the number in the image, size of the image, angle of the number, etc. These factors are independent of each other.
+
+Feature disentanglement involves separating underlying concepts of “Big one in the left”: ie: size(big), number(one), location(left).
+Our interest here is to see if we can isolate these factors in the latent code so that we can have control over the generation of the images. So we want the encoder to disentangle the representation into different factors and then we generate the image with desired factors say “small seven in the top rotated 30 degrees”.
+
+<figure>
+<center>
+  <img width="800" height="400" alt="Can't See? Something went wrong!" src="https://d3i71xaburhd42.cloudfront.net/35da0a2001eea88486a5de677ab97868c93d0824/6-Figure2-1.png"/>
+  <figcaption>Fig 5: Generated MNIST images by InfoGAN [5] varied digit, thickness and roatation.</figcaption>
+</center>
+</figure>
+
+
+## Beta-VAE
+Beta-VAE is a variant of VAE which allows disentanglement of the learned latent code. Beta-VAE adds hyperparameter to the loss function which modulates the learning constraint of VAE.
+
+<figure>
+<center>
+  <img width="800" alt="Can't See? Something went wrong!" src="https://miro.medium.com/max/1400/1*Z6tj5bVoArekVgv65gfkfg.png"/>
+  <figcaption>Fig 6: Loss function of beta-VAE [7].</figcaption>
+</center>
+</figure>
+
+The first part of the loss function takes care of the reconstruction of the image, it is the second term that learns the latent code of VAE. The advantage of using  Gaussians with covariant matrices is that different dimensions that span across are independent. So increasing the weight of this(second part) loss, makes the latent code to be disentangled and independent. But this also brings a tradeoff between disentanglement and the reconstruction capability of the VAE.
+
+<figure>
+<center>
+  <img width="800" alt="Can't See? Something went wrong!" src="https://production-media.paperswithcode.com/methods/Screen_Shot_2020-06-28_at_4.00.13_PM.png"/>
+  <figcaption>Fig 7: Samples generated by beta-VAE [7].</figcaption>
+</center>
+</figure>
+
+
+
+
+## Beta-TCVAE
+beta-TCVAE decomposes the KL divergence[10] term of the loss function of VAE into reconstruction loss, Index-code mutual information[8] between data and latent variable, Total Correlation[9] of z, and Dimension wise KL divergence[10] of $$z$$(respectively in the following formula). This helps to break the overall KL Divergence of $$z$$ into dimension-wise quantities, which will focus on each dimension of the latent code $$z$$. In this formulation, the beta hyperparameter is only on the Total Correlation term which is more important for disentanglement without affecting the reconstruction.
```

Comment:
What does Beta-TCVAE help achieve as compared to Beta-VAE's that help in disentanglement ?

Reply:
added this point in both beta VAE and beta TCVAE - because beta VAE doesn't have good reconstruction